cmd/relay/bgs/bgs.go

```diff
@@ -0,0 +1,1376 @@
+			return err
+		}
+
+		return nil
+	case env.RepoTombstone != nil:
+		// TODO: DEPRECATED - expect account message above instead
+		if err := bgs.handleRepoTombstone(ctx, host, env.RepoTombstone); err != nil {
+			return err
+		}
+
+		return nil
+	default:
+		return fmt.Errorf("invalid fed event")
+	}
+}
+
+func (bgs *BGS) handleCommit(ctx context.Context, host *models.PDS, evt *comatproto.SyncSubscribeRepos_Commit) error {
+	bgs.log.Debug("bgs got repo append event", "seq", evt.Seq, "pdsHost", host.Host, "repo", evt.Repo)
+
+	//s := time.Now()
+	u, err := bgs.lookupUserByDid(ctx, evt.Repo)
+	//userLookupDuration.Observe(time.Since(s).Seconds())
+	if err != nil {
+		if !errors.Is(err, gorm.ErrRecordNotFound) {
+			repoCommitsResultCounter.WithLabelValues(host.Host, "nou").Inc()
+			return fmt.Errorf("looking up event user: %w", err)
+		}
+
+		newUsersDiscovered.Inc()
+		start := time.Now()
+		subj, err := bgs.createExternalUser(ctx, evt.Repo, host)
+		newUserDiscoveryDuration.Observe(time.Since(start).Seconds())
+		if err != nil {
+			repoCommitsResultCounter.WithLabelValues(host.Host, "uerr").Inc()
+			return fmt.Errorf("fed event create external user: %w", err)
+		}
+
+		u = subj
+	}
+
+	ustatus := u.GetUpstreamStatus()
+	//span.SetAttributes(attribute.String("upstream_status", ustatus))
+
+	if u.GetTakenDown() || ustatus == events.AccountStatusTakendown {
+		//span.SetAttributes(attribute.Bool("taken_down_by_relay_admin", u.GetTakenDown()))
+		bgs.log.Debug("dropping commit event from taken down user", "did", evt.Repo, "seq", evt.Seq, "pdsHost", host.Host)
+		repoCommitsResultCounter.WithLabelValues(host.Host, "tdu").Inc()
+		return nil
+	}
+
+	if ustatus == events.AccountStatusSuspended {
+		bgs.log.Debug("dropping commit event from suspended user", "did", evt.Repo, "seq", evt.Seq, "pdsHost", host.Host)
+		repoCommitsResultCounter.WithLabelValues(host.Host, "susu").Inc()
+		return nil
+	}
+
+	if ustatus == events.AccountStatusDeactivated {
+		bgs.log.Debug("dropping commit event from deactivated user", "did", evt.Repo, "seq", evt.Seq, "pdsHost", host.Host)
+		repoCommitsResultCounter.WithLabelValues(host.Host, "du").Inc()
+		return nil
+	}
+
+	if evt.Rebase {
+		repoCommitsResultCounter.WithLabelValues(host.Host, "rebase").Inc()
+		return fmt.Errorf("rebase was true in event seq:%d,host:%s", evt.Seq, host.Host)
+	}
+
+	if host.ID != u.PDS && u.PDS != 0 {
+		bgs.log.Warn("received event for repo from different pds than expected", "repo", evt.Repo, "expPds", u.PDS, "gotPds", host.Host)
+		// Flush any cached DID documents for this user
+		bgs.purgeDidCache(ctx, evt.Repo)
+
+		subj, err := bgs.createExternalUser(ctx, evt.Repo, host)
+		if err != nil {
+			repoCommitsResultCounter.WithLabelValues(host.Host, "uerr2").Inc()
+			return err
+		}
+
+		if subj.PDS != host.ID {
+			repoCommitsResultCounter.WithLabelValues(host.Host, "noauth").Inc()
+			return fmt.Errorf("event from non-authoritative pds")
+		}
+	}
+
+	if u.GetTombstoned() {
```

Comment:
I'm pretty sure we just don't / shouldn't have a concept of "tombstoned" accounts in the relay. If the DID is tombstoned, that is basically account status `deleted`

Reply:
yeah, old code, I basically haven't had time to update the account status flow yet, that part seemed to have been in flux up until last week